app/core/rate_limiting/embedding_limiter.py

```
@@ -0,0 +1,141 @@
+"""
+Rate limiter for Google Gemini embedding API
+
+Implements rate limiting for the free tier of gemini-embedding-001:
+- 100 RPM (requests per minute)
+- 30,000 TPM (tokens per minute)  
+- 1,000 RPD (requests per day)
+"""
+
+import asyncio
+import time
+from typing import Optional, Dict, Any
```

Ruff (0.12.2)

`typing.Optional` imported but unused

Remove unused import: `typing.Optional`

(F401)